File: Samples/FullScreenMode/cppwinrt/Scenario1_Basic.cpp
//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
// This code is licensed under the MIT License (MIT).
// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
//
//*********************************************************

#include "pch.h"
#include "Scenario1_Basic.h"
#include "Scenario1_Basic.g.cpp"

using namespace winrt;
using namespace Windows::Foundation;
using namespace Windows::UI::ViewManagement;
using namespace Windows::UI::Xaml;
using namespace Windows::UI::Xaml::Controls;
using namespace Windows::UI::Xaml::Input;
using namespace Windows::UI::Xaml::Navigation;

namespace winrt::SDKTemplate::implementation
{
    Scenario1_Basic::Scenario1_Basic()
    {
        InitializeComponent();
    }

    void Scenario1_Basic::OnNavigatedTo(NavigationEventArgs const&)
    {
        windowResizeToken = Window::Current().SizeChanged([this](auto&&, auto&&)
        {
            UpdateContent();
        });
        keyDownToken = rootPage.KeyDown({ this, &Scenario1_Basic::RootPage_OnKeyDown });

        UpdateContent();
    }

    void Scenario1_Basic::OnNavigatedFrom(NavigationEventArgs const&)
    {
        Window::Current().SizeChanged(windowResizeToken);
        rootPage.KeyDown(keyDownToken);
    }

    void Scenario1_Basic::ToggleFullScreenModeButton_Click(IInspectable const&, RoutedEventArgs const&)
    {
        ApplicationView view = ApplicationView::GetForCurrentView();
        if (view.IsFullScreenMode())
        {
            view.ExitFullScreenMode();
            rootPage.NotifyUser(L"Exiting full screen mode", NotifyType::StatusMessage);
            isLastKnownFullScreen = false;
            // The SizeChanged event will be raised when the exit from full screen mode is complete.
        }
        else
        {
            if (view.TryEnterFullScreenMode())
            {
                rootPage.NotifyUser(L"Entering full screen mode", NotifyType::StatusMessage);
                isLastKnownFullScreen = true;
                // The SizeChanged event will be raised when the entry to full screen mode is complete.
            }
            else
            {
                rootPage.NotifyUser(L"Failed to enter full screen mode", NotifyType::ErrorMessage);
            }
        }
    }

    void Scenario1_Basic::ShowStandardSystemOverlaysButton_Click(IInspectable const&, RoutedEventArgs const&)
    {
        ApplicationView view = ApplicationView::GetForCurrentView();
        view.ShowStandardSystemOverlays();
    }

    void Scenario1_Basic::UseMinimalOverlaysCheckBox_Click(IInspectable const&, RoutedEventArgs const&)
    {
        ApplicationView view = ApplicationView::GetForCurrentView();
        view.FullScreenSystemOverlayMode(UseMinimalOverlaysCheckBox().IsChecked().Value() ? FullScreenSystemOverlayMode::Minimal : FullScreenSystemOverlayMode::Standard);
    }

    void Scenario1_Basic::UpdateContent()
    {
        ApplicationView view = ApplicationView::GetForCurrentView();
        bool isFullScreenMode = view.IsFullScreenMode();
        ToggleFullScreenModeSymbol().Symbol(isFullScreenMode ? Symbol::BackToWindow : Symbol::FullScreen);
        ReportFullScreenMode().Text(isFullScreenMode ? L"is in" : L"is not in");
        FullScreenOptionsPanel().Visibility(isFullScreenMode ? Visibility::Visible : Visibility::Collapsed);

        // Did the system force a change in full screen mode?
        if (isLastKnownFullScreen != isFullScreenMode)
        {
            isLastKnownFullScreen = isFullScreenMode;
            // Clear any stray messages that talked about the mode we are no longer in.
            rootPage.NotifyUser(L"", NotifyType::StatusMessage);
        }
    }

    void Scenario1_Basic::RootPage_OnKeyDown(IInspectable const&, KeyRoutedEventArgs const& e)
    {
        if (e.Key() == Windows::System::VirtualKey::Escape)
        {
            ApplicationView view = ApplicationView::GetForCurrentView();
            if (view.IsFullScreenMode())
            {
                view.ExitFullScreenMode();
                rootPage.NotifyUser(L"Exited full screen mode due to keypress", NotifyType::StatusMessage);
                isLastKnownFullScreen = false;
            }
        }
    }
}

File: archived/FullScreenMode/js/js/scenario1-basic.js
//// Copyright (c) Microsoft Corporation. All rights reserved

(function () {
    "use strict";

    var ViewManagement = Windows.UI.ViewManagement;
    var FullScreenSystemOverlayMode = ViewManagement.FullScreenSystemOverlayMode;
    var ApplicationView = ViewManagement.ApplicationView;

    // What is the program's last known full-screen state?
    // We use this to detect when the system forced us out of full-screen mode.
    var isLastKnownFullScreen = ApplicationView.getForCurrentView().isFullScreenMode;

    var page = WinJS.UI.Pages.define("/html/scenario1-basic.html", {
        ready: function (element, options) {
            toggleFullScreenMode.addEventListener("click", onToggleFullScreenMode);
            showStandardSystemOverlays.addEventListener("click", onShowStandardSystemOverlays);
            useMinimalOverlays.addEventListener("check", onMinimalOverlaysChanged);
            // The resize event is raised when the view enters or exits full screen mode.
            window.addEventListener("resize", onresize);
            document.body.addEventListener("keydown", onkeydown);
            onresize();
        },

        unload: function () {
            window.removeEventListener("resize", onresize);
            document.body.removeEventListener("keydown", onkeydown);
        }
    });

    function onToggleFullScreenMode() {
        var view = ApplicationView.getForCurrentView();
        if (view.isFullScreenMode) {
            view.exitFullScreenMode();
            WinJS.log && WinJS.log("Exiting full screen mode", "samples", "status");
            isLastKnownFullScreen = false;
            // The resize event will be raised when the exit from full screen mode is complete.
        } else {
            if (view.tryEnterFullScreenMode()) {
                WinJS.log && WinJS.log("Entering full screen mode", "samples", "status");
                isLastKnownFullScreen = true;
                // The resize event will be raised when the entry to full screen mode is complete.
            } else {
                WinJS.log && WinJS.log("Failed to enter full screen mode", "samples", "error");
            }    
        }
    }
    
    function onShowStandardSystemOverlays() {
        var view = ApplicationView.getForCurrentView();
        view.showStandardSystemOverlays();
    }

    function onMinimalOverlaysChanged() {
        var view = ApplicationView.getForCurrentView();
        view.fullScreenSystemOverlayMode = useMinimalOverlays.checked ? FullScreenSystemOverlayMode.minimal : FullScreenSystemOverlayMode.standard;
    }

    function onresize() {
        var view = ApplicationView.getForCurrentView();
        var isFullScreenMode = view.isFullScreenMode;
        toggleFullScreenMode.innerText = WinJS.UI.AppBarIcon[isFullScreenMode ? "backtowindow" : "fullscreen"];
        reportFullScreenMode.innerText = isFullScreenMode ? "is in" : "is not in";
        fullScreenOptions.style.display = isFullScreenMode ? "block" : "none";

        // Did the system force a change in full screen mode?
        if (isLastKnownFullScreen !== isFullScreenMode) {
            isLastKnownFullScreen = isFullScreenMode;
            // Clear any stray messages that talked about the mode we are no longer in.
            WinJS.log && WinJS.log("", "samples", "status");
        }

    }

    function onkeydown(e) {
        if (e.key === "Escape") {
            var view = ApplicationView.getForCurrentView();
            if (view.isFullScreenMode) {
                view.exitFullScreenMode();
                WinJS.log && WinJS.log("Exited full screen mode due to keypress", "samples", "status");
                isLastKnownFullScreen = false;
            }
        }
    }
})();
